### ISSUE TITLE
Hover shows type of `Unknown` for object passed as value to unspecialized `dict`

### ISSUE THREAD

**Issue by debonte** [microsoft/pyright#5422]
**Describe the bug**
When calling a function that takes a `dict` without type arguments, hovering over objects passed as values in that dictionary shows a type of `Unknown`. This may make sense from a type checking perspective, but seems unfortunate from a language server perspective.

**To Reproduce**
```python
value = "str"

def func(string: str, dictionary: dict): ...

def caller():
    func(value,          # hover reveals "Literal['str']"
         { "x": value }) # hover reveals "Unknown"
```

**Expected behavior**
Hovering over `value` in `{ "x": value }` reveals a type of `Literal['str']` just like when `value` is passed outside of the dictionary (first argument).

**VS Code extension or command-line**
Pyright 1.1.316 with VS Code 1.80.0-insiders

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.317, which I just published. It will also be included in future releases of pylance.
